app/openai_ops.py

```diff
@@ -359,76 +359,98 @@ def context_length(
         raise NotImplementedError(error)
 
 
-# Adapted from https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
```

Comment:
please don't completely delete this comment; you can say "initially adapted from ... and then we customized it to support image file scenarios" or something like that